crates/api_common/src/utils.rs

```diff
@@ -355,6 +355,16 @@ pub fn check_private_instance(
   }
 }
 
+/// If private messages are disabled, dont allow them to be sent / received
+#[tracing::instrument(skip_all)]
+pub fn check_private_messages_enabled(local_user_view: &LocalUserView) -> Result<(), LemmyError> {
+  if !local_user_view.local_user.enable_private_messages {
+    Err(LemmyErrorType::CouldntCreatePrivateMessage)?
+  } else {
+    Ok(())
+  }
```

Comment:
This can be expressed in a different way that you may or may not prefer:

```rust
  local_user_view
    .local_user
    .enable_private_messages
    .then_some(())
    .ok_or_else(|| LemmyErrorType::CouldntCreatePrivateMessage.into())
```

Reply:
Not a huge fan of that one at least in this case, because it unnecessarily coerces it into an option, when it's already a bool.